common/crypto/verify_ias_report/verify-report.cpp

```diff
@@ -58,6 +61,39 @@ static int EVP_DecodeBlock_wrapper(unsigned char* out, int out_len, const unsign
 //########### INTERNAL FUNCTIONS #########################################
 //>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
 
+#define IAS_QUOTE_STATUS_JSON_STRING "isvEnclaveQuoteStatus"
+
+const char* quote_status[QS_NUMBER] = {"bad status", "OK", "GROUP_OUT_OF_DATE",
+    "CONFIGURATION_NEEDED", "SW_HARDENING_NEEDED", "CONFIGURATION_AND_SW_HARDENING_NEEDED"};
+
+quote_status_e get_quote_status(const char* ias_report, unsigned int ias_report_len)
+{
+    JSON_Value* jv;
+    JSON_Object* jo;
+    const char* s;
+    int i;
+
+    jv = json_parse_string(ias_report);
+    COND2ERR(jv == NULL);
+
+    jo = json_value_get_object(jv);
+    COND2ERR(jo == NULL);
+
+    s = json_object_get_string(jo, IAS_QUOTE_STATUS_JSON_STRING);
+    COND2ERR(s == NULL);
+
+    for (i = 1; i < QS_NUMBER; i++)
+    {
+        if (0 == strncmp(s, quote_status[i], strlen(quote_status[i])))
+        {
```

Comment:
strlen has no termination.

Reply:
That's a constant string.
Anyway I changed that but the code relies on Parson to null-terminate the returned string.
New versions of Parson (if we want to upgrade it) give the possibility to get the string length.